chcemvediet/apps/inforequests/admin.py

```diff
@@ -384,26 +376,24 @@ def get_queryset(self, request):
     def get_inforequest(self, obj):
         return obj.branch.inforequest
 
-    def has_delete_permission(self, request, obj=None):
-        if obj is None:
-            return True
+    def delete_dependency(self, obj):
+        dependency = []
         if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
-            return False
-        if len(obj.branch.actions) > 1:
-            return True
-        return False
+            dependency.append(format_html(
+                u'{} is type {}.'.format(admin_obj_format(obj), obj.get_type_display())))
+        if len(obj.branch.actions) == 1:
+            dependency.append(format_html(
+                u'{} is the only action in the branch.'.format(admin_obj_format(obj))))
+        return dependency
 
     def render_delete_form(self, request, context):
-        action = context[u'object']
-        if not action.is_last_action:
-            context[u'deleted_objects'].insert(0, format_html(squeeze(u"""
-                <b>Warning:</b> The deleted action is not the last action in the branch. Deleting it
-                may cause logical errors in the inforequest history.
-                """)))
+        context[u'delete_dependency'] = self.delete_dependency(context[u'object'])
         context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
         return super(ActionAdmin, self).render_delete_form(request, context)
 
     def delete_model(self, request, obj):
+        if self.delete_dependency(obj):
+            raise PermissionDenied
```

Comment:
@martinmacko47 Ak raisnem `PermissionDenied` prehliadac ma upozrni na pomaly response. Ak som vyhodil vynimku napr `Http404` tak to bolo OK.

```
Slow server response time (1.95s). The recommended limit is 800ms.
```

Reply:
@viliambalaz Ten slow response je sposobeny zrejme nacitavanim debug informacii do error hlasky. To nevadi, ze pri developovani su debug obrazovky pomale.

Vacsi problem je, ze (aspon teda mne) to skonci errorom:

![error](https://user-images.githubusercontent.com/5547275/112755665-f2de4a80-8fe1-11eb-9a27-955f3a94e93e.png)

Vyzera, ze to opat pada na `{% change_lang lang.0 %}` ako je na tom screenshote zvyraznene. Ked danu cast zakomentujem, tak to skonci na spravnej 403 obrazovke:

![403](https://user-images.githubusercontent.com/5547275/112755693-228d5280-8fe2-11eb-84e4-d0799e4346ce.png)

Zislo by za oddebugovat, v com je problem s tym `{% change_lang lang.0 %}` zase.

Reply:
Hm. Vyzera, ze staci odchytit aj `NoReverseMatch` v `change_lang`:

```
    except (Resolver404, NoReverseMatch):
        url = path
```

Reply:
Vytvoril som samostatny PR na riesenie tohoto bugu. PR: https://github.com/martinmacko47/chcemvediet/pull/362

Reply:
:ok: